pkg/postgrest/api.go

```diff
@@ -0,0 +1,107 @@
+package postgrest
```

Comment:
I think it’s better to encapsulate this layer as an API client, keeping PostgREST as an internal concept to allow for future migration.